examples/technical_details/plot_sklearn_api.py

```diff
@@ -0,0 +1,170 @@
+"""
+.. _example_sklearn_api:
+
+=====================================================================================
+Using skore with scikit-learn compatible estimators
+=====================================================================================
+
+This example shows how to use skore with scikit-learn compatible estimators.
+For these models, skore provides a unified interface for reporting and evaluation.
+Any model that can be used with the scikit-learn API can be used with skore.
+Skore's :class:`~skore.EstimatorReport` can be used to report on any estimator
+that has a ``fit`` and ``predict`` method.
+In fact, skore only requires the ``predict`` method if the estimator has already
+been fitted.
+
+.. note::
+
+  When using an :class:`~skore.EstimatorReport` to compute the ROC AUC and ROC curve
+  for a classification task, the estimator must have a ``predict_proba`` method.
+
+In this example, we showcase a gradient boosting model
+(`XGBoost <https://github.com/dmlc/xgboost>`_) and a custom estimator.
+
+.. note::
+
+    This example is not exhaustive; many other scikit-learn compatible models can be
+    used with skore:
+
+    -   More gradient boosting libraries like
+        `LightGBM <https://github.com/microsoft/LightGBM>`_, and
+        `CatBoost <https://github.com/catboost/catboost>`_,
+
+    -   Deep learning frameworks such as `Keras <https://github.com/keras-team/keras>`_
+        and `skorch <https://github.com/skorch-dev/skorch>`_ (a wrapper for
+        `PyTorch <https://github.com/pytorch/pytorch>`_).
+
+    -   Tabular foundation models such as
+        `TabICL <https://github.com/soda-inria/tabicl>`_ and
+        `TabPFN <https://github.com/PriorLabs/TabPFN>`_,
+
+    -   etc.
+
+"""
+
+# %%
+# Loading a binary classification dataset
+# =======================================
+#
+# We generate a synthetic binary classification dataset with only 1,000 samples to keep
+# the computation time reasonable:
+
+# %%
+from sklearn.datasets import make_classification
+
+X, y = make_classification(
+    n_samples=1_000,
+    n_features=5,
+    n_classes=2,
+    class_sep=0.3,
+    n_clusters_per_class=1,
+    random_state=42,
+)
+print(f"{X.shape = }")
+
+# %%
+# We split our data:
+
+# %%
+from skore import train_test_split
+
+split_data = train_test_split(X, y, shuffle=True, random_state=42, as_dict=True)
```

Comment:
why is shuffle explicitely True, since it's the default, plus it triggers a warning?  Let's either remove it, or pass it to False.